sympy/core/expr.py

```diff
@@ -1940,6 +1940,21 @@ def extract_multiplicatively(self, c):
            x/6
 
         """
+        if (type(c) == int and c < 0):
+            return Expr.extract_multiplicatively(self, c*S.ImaginaryUnit*-S.ImaginaryUnit)
```

Comment:
To convert to Integer you could do `Integer(c)`. But I can't see how `c*I - I` will ever equal `c` which means this code is never going to work properly for the query intended...unless I am totally missing something.